src/app/openapi.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "ccf/app_interface.h"
+
+#include <vector>
+
+namespace app::openapi
```

Comment:
I think a comment here to say that this is inevitable duplication and needs to be kept in sync with `.proto` files would be useful.